### PR TITLE
Margin data endpoints

### PR DESCRIPTION
**Pull request by TomaszGol**
## Summary by CodeRabbit

- **New Features**
  - Introduced the ability to fetch and display margin data for specific markets and wallet addresses.

- **Enhancements**
  - Added new utility functions for encoding and converting values to improve flexibility and utility.

- **API Integrations**
  - Integrated new API calls to fetch margin data using `HttpProvider` and handle blockchain interactions.

### CODE REVIEW

File: src/server-extension/resolvers/marginDataResolver.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+@Resolver()
+export class MarginDataResolver {
+  @Query(() => MarginData)
+  async getMarginData(
+    @Arg("marketId", { nullable: false }) marketId: string,
+    @Arg("walletAddress", { nullable: false }) walletAddress: string,
+  ): Promise<MarginData> {
+    if (marketId.length === 0 || walletAddress.length === 0) {
```
Comment (Klapeyron) [inline src/server-extension/resolvers/marginDataResolver.ts:22]: we can check if marketId is an integer (unsigned 64 bit) and if walletAddress is ss58

File: src/server-extension/resolvers/marginDataResolver.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  }
+}
+
+@Resolver()
+export class MarginDataResolver {
+  @Query(() => MarginData)
+  async getMarginData(
+    @Arg("marketId", { nullable: false }) marketId: string,
+    @Arg("walletAddress", { nullable: false }) walletAddress: string,
+  ): Promise<MarginData> {
```
Comment (Klapeyron) [inline src/server-extension/resolvers/marginDataResolver.ts:21]: MarginData is not a string

File: src/utils/encodersUtils.ts [hunk 3]
```diff
@@ -35,3 +35,49 @@ export const convertStringValueToHexBigEndian = (value: string) => {
...
+  const littleEndianBytes = hexBytes.reverse();
+  return "0x" + littleEndianBytes.join("");
+};
+
+export const convertToU8a = async (api: any, address: string) => {
+  const decodedAddress = api.createType("AccountId", address);
+
+  const u8aRepresentation = decodedAddress.toU8a(true);
+  return bufferToHex(u8aRepresentation);
+};
```
Comment (Klapeyron) [inline src/utils/encodersUtils.ts:83]: why do we need this functions?

Comment (TomaszGol) [reply src/utils/encodersUtils.ts]: Changed it as I thought we need to encode values properly, but turned out it is overkill, adjusted it a little

**Review by Klapeyron — APPROVED** [review]

**Pull request merged**